src/app/pages/account/edit/[username].vue

```diff
@@ -64,44 +189,96 @@ if (route.params.username !== store.signedInUsername) {
 }
 
 // api data
-const accountDeleteMutation = useAccountDeleteMutation()
 const accountByUsernameQuery = useAccountByUsernameQuery({
   username: route.params.username,
 })
-const profilePictureSetMutation = useProfilePictureSetMutation()
-const api = await useApiData([
-  accountByUsernameQuery,
-  profilePictureSetMutation,
-])
-const account = getAccountItem(api.value.data.accountByUsername)
+
+const profilePictureQuery = useProfilePictureByAccountIdQuery({
+  accountId: store.signedInAccountId,
+})
+
+const api = await useApiData([accountByUsernameQuery, profilePictureQuery])
+
+const account = computed(() => getAccountItem(api.value.data.accountByUsername))
 
 // methods
 const onUploadSelect = async (uploadId?: string | null | undefined) =>
   await profilePictureSetMutation.executeMutation({
     uploadId,
   })
+
 const showModalUploadSelection = () => {
   store.modals.push({ id: 'ModalUploadSelection' })
 }
+
+const toggleEdit = () => {
+  if (!isEditing.value) {
+    editableDescription.value = account.value?.description?.trim() || ''
+  }
+  isEditing.value = !isEditing.value
+}
+
+const cancelEdit = () => {
+  editableDescription.value = account.value?.description?.trim() || ''
+  isEditing.value = false
+}
+
+const openDeleteDrawer = () => {
+  isDeleteDrawerOpen.value = true
+}
+
+const saveDescription = async () => {
+  if (!account.value?.username) return
+
+  const result = await accountDescriptionMutation.executeMutation({
+    username: account.value.username,
+    accountPatch: { description: editableDescription.value },
+  })
+
+  if (!result.error) {
+    isEditing.value = false
+  }
+}
+
+const removeProfilePicture = async () => {
+  const profilePicture = getProfilePictureItem(
+    api.value.data.profilePictureByAccountId,
+  )
+  const upload = getUploadItem(profilePicture?.uploadByUploadId)
+
+  if (!upload?.id) return
+
+  await deleteUploadMutation.executeMutation({
```

Comment:
Deleting a profile picture should not delete the upload, but from the profile picture table only.

(or ask if the upload should be deleted as well, but that's not for now I guess)